File: syno_backend/src/main/java/com/syno_back/backend/dto/UpdateUserDictionaryDto.java
package com.syno_back.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.List;

/**
 * DTO for updating <code>DbUserDictionary</code>
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
@ToString
public class UpdateUserDictionaryDto {
    /**
     * Unique id for <code>DbUserDicionary</code>
     */
    @JsonProperty("pin")
    private String pin;

    /**
     * Dictionary name
     */
    @Size(max=100)
    @JsonProperty("name")
    private String name;

    /**
     * Dictionary language
     */
    @Size(max=100)
    @JsonProperty("language")
    private String language;

    /**
     * Time when dictionary was created
     */
    @JsonProperty("time_created")
    private LocalDateTime timeCreated;

    /**
     * Time when dictionary was modified
     */
    @JsonProperty("time_modified")
    private LocalDateTime timeModified;

    /**
     * List of dictionary cards
     */
    @JsonProperty("user_cards")
    private List<UpdateUserCardDto> userCards;
}

File: syno-mobile/syno-mobile/Core/Dtos/UpdateUserDictionaryDto.swift
import Foundation

///DTO for updating `DbUserDictionary` on server
class UpdateUserDictionaryDto: Codable {
    let pin: String
    
    let name: String
    
    let language: String
    
    let timeCreated: Date?
    
    let timeModified: Date?
    
    let userCards: [UpdateUserCardDto]
    
    /**
     Creates new `UpdateUserDictionaryDto`
     - Parameters:
        - pin: unique `DbUserDictionary` id
        - name: dictionary name
        - language: dictionary language
        - timeCreated: time when `DbUserDictionary` was created
        - timeModified: time when `UpdateUserDictionaryDto` was modified
        - userCards: list of dictionaries cards
     */
    init(pin: String, name: String, language: String, timeCreated: Date?, timeModified: Date?, userCards: [UpdateUserCardDto]) {
        self.pin = pin
        self.name = name
        self.language = language
        self.timeCreated = timeCreated
        self.timeModified = timeModified
        self.userCards = userCards
    }
}
